lib/Sema/CSApply.cpp

```diff
@@ -8148,6 +8149,95 @@ bool ConstraintSystem::applySolutionFix(
                                       fix.first.getArgumentLabels(*this),
                                       isa<SubscriptExpr>(call->getFn()));
   }
+
+  case FixKind::AddConformace: {
+    auto getMissingConformance = [&](ConstraintLocator *locator) {
+      auto *anchor = locator->getAnchor();
+      auto &requirement = locator->getPath().back();
+      auto result = MissingConformances.find({anchor, requirement.getValue()});
+      assert(result != MissingConformances.end());
+      return result->getSecond();
+    };
+
+    auto conformance = getMissingConformance(locator);
+
+    auto *anchor = locator->getAnchor();
+    auto owner = solution.simplifyType(getType(anchor));
+
+    auto type = conformance.first;
+    auto protocolType = conformance.second->getDeclaredType();
+
+    //  Find `ApplyExpr` based on a function expression attached to it.
+    auto findApplyExpr = [](Expr *parent, Expr *fnExpr) -> ApplyExpr * {
```

Comment:
I'm a little surprised that we have to go searching for the `ApplyExpr` and matching up the parameter. Doesn't the (pre-simplified) locator contain the `ApplyExpr`, as well as information about which parameter we're talking about? Or is that still hard to get to because of the matchTypes/function call mess?

Reply:
Yes, unfortunately that exactly the reason, currently generic requirements are opened from function expression, which means that it would have `DeclRefExpr` or `UnresolvedDotExpr` or `OverloadedDeclRefExpr` as its anchor and `opened generic -> type requirement #` as its path...